### DISCUSSION TITLE
Warning: Code for IBM i will not function correctly until the temporary library has been corrected in the settings.

### DISCUSSION THREAD

**Discussion by n8bush** [codefori/vscode-ibmi#1853] [Q&A]
Updated to 2.7 and am receiving the message in the title when connecting to my development environment. Connecting to one of our customers machines works fine, so feel like this is some misconfiguration/change on my dev machine. Hoping someone can point me in the right direction.

QCCSID is 37, 
QIBM_PASE_CCSID is 1208, 
Bash is installed from yum,
which bash /QOpenSys/pkgs/bin/bash,
/QOpenSys/QIBM/ProdData/SC1/OpenSSH/etc/sshd_config has: ibmpaseforishell /QOpenSys/pkgs/bin/bash
$PATH=/QIBM/ProdData/Access/ACS/Base:/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit/bin:/QOpenSys/pkgs/lib/nodejs18/bin:/QOpenSys/pkgs/bin:/QIBM/ProdData/Access/ACS/Base:/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit/bin:/QOpenSys/pkgs/lib/nodejs18/bin:/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin:/usr/local/bin:/QOpenSys/pkgs/python3.9/site-packages:/usr/local/bin:/QOpenSys/pkgs/python3.9/site-packages

.: echo "This should be the only text!"
{
    "code": 0,
    "signal": null,
    "stdout": "This should be the only text!",
    "stderr": ""
}

.: echo $HOME && cd && test -w $HOME
{
    "code": 0,
    "signal": null,
    "stdout": "/home/myuser",
    "stderr": ""
}

/home/myuser: /QOpenSys/usr/bin/qsh
liblist
{
    "code": 127,
    "signal": null,
    "stdout": "",
    "stderr": "bash: /QOpenSys/usr/bin/qsh: No such file or directory"
}

/home/myuser: /QOpenSys/usr/bin/qsh
system  "CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')"
{
    "code": 127,
    "signal": null,
    "stdout": "",
    "stderr": "bash: /QOpenSys/usr/bin/qsh: No such file or directory"
}
[code for IBM i.txt](https://github.com/codefori/vscode-ibmi/files/14334878/code.for.IBM.i.txt)

**Comment by n8bush** [comment]
Issue was I was using a misconfigured ssh tunnel to access development, switched to VPN.

**Comment by chrjorgensen** [comment]
@n8bush Glad you found the issue. 👍
